### PR TITLE
kernel: Always build the kernel version found locally for CI

### PR DESCRIPTION
**Pull request by ganeshmaharaj**
Fixes: #1441

Signed-off-by: Ganesh Maharaj Mahalingam <ganesh.mahalingam@intel.com>
(cherry picked from commit 16cf741ca60e3931f0b798ab035336694b76fbc5)

### CODE REVIEW

**Comment by ganeshmaharaj** [comment]
@chavafg

File: .ci/install_kata_kernel.sh [hunk 1]
```diff
@@ -65,11 +65,12 @@ get_packaged_kernel_version() {
 }
 
 build_and_install_kernel() {
+	# Always build and install the kernel version found locally
 	info "Install kernel from sources"
 	pushd "${tmp_dir}" >> /dev/null
-	"${kernel_repo_dir}/kernel/build-kernel.sh" "setup"
-	"${kernel_repo_dir}/kernel/build-kernel.sh" "build"
-	sudo -E PATH="$PATH" "${kernel_repo_dir}/kernel/build-kernel.sh" "install"
+	"${kernel_repo_dir}/kernel/build-kernel.sh" -v "${kernel_version}" "setup"
```
Comment (grahamwhaley) [inline .ci/install_kata_kernel.sh]: I know this is a cherry pick to a stable backport, but whilst reading it I noticed that this func is using the value of `kernel_version`, which is a var that seems to be declared and used in multiple functions in this file - I suspect either:
 - we should make that an explicit global var
 - or we should make the ones already present explicitly function local vars, and make them all call the version getting func.

Comment (ganeshmaharaj) [reply .ci/install_kata_kernel.sh]: +1. Will do a follow-on cleanup PR around all this.

**Comment by ganeshmaharaj** [comment]
The one failure seems to be around CPUs again.

```
08:42:34 [1] • Failure [14.179 seconds]
08:42:34 [1] Update CPU constraints
08:42:34 [1] /tmp/jenkins/workspace/kata-containers-tests-ubuntu-18-04-PR/go/src/github.com/kata-containers/tests/integration/docker/cpu_test.go:324
08:42:34 [1]   Update CPU set
08:42:34 [1]   /tmp/jenkins/workspace/kata-containers-tests-ubuntu-18-04-PR/go/src/github.com/kata-containers/tests/vendor/github.com/onsi/ginkgo/extensions/table/table.go:92
08:42:34 [1]     cpuset should be equal to 0 [It]
08:42:34 [1]     /tmp/jenkins/workspace/kata-containers-tests-ubuntu-18-04-PR/go/src/github.com/kata-containers/tests/vendor/github.com/onsi/ginkgo/extensions/table/table_entry.go:46
08:42:34 [1] 
08:42:34 [1]     Expected
08:42:34 [1]         <int>: 126
08:42:34 [1]     to be zero-valued
08:42:34 [1] 
08:42:34 [1]     /tmp/jenkins/workspace/kata-containers-tests-ubuntu-18-04-PR/go/src/github.com/kata-containers/tests/integration/docker/cpu_test.go:395
08:42:34 [1] ------------------------------
08:42:34 [1] SSSSSSSSSSSSSSSSSSSSS
08:42:34 [1] 
08:42:34 [1] Summarizing 1 Failure:
08:42:34 [1] 
08:42:34 [1] [Fail] Update CPU constraints Update CPU set [It] cpuset should be equal to 0 
08:42:34 [1] /tmp/jenkins/workspace/kata-containers-tests-ubuntu-18-04-PR/go/src/github.com/kata-containers/tests/integration/docker/cpu_test.go:395
08:42:34 [1] 
08:42:34 [1] Ran 45 of 68 Specs in 1075.471 seconds
08:42:34 [1] FAIL! -- 44 Passed | 1 Failed | 0 Pending | 23 Skipped --- FAIL: TestIntegration (1075.48s)
```

**Comment by devimc** [comment]
@ganeshmaharaj this is a real issue in libcontainer. I thought it was fixed

```
Stdout: OCI runtime exec failed: rpc error: code = Internal desc = Could not run process: container_linux.go:348: starting container process caused "process_linux.go:90: adding pid 128 to cgroups caused \"failed to write 128 to cgroup.procs: write /sys/fs/cgroup/cpu,cpuacct/docker/99144fb4fbe891965a8e950515949fca825d362a936e93a8383c798e8dbefb70/cgroup.procs: invalid argument\"": unknown
```

cc @chavafg @sboeuf

**Comment by devimc** [comment]
ohh wait, libcontainer was not updated in the agent stable-1.6

**Pull request merged**